gfx/layers/client/TiledContentClient.cpp

```diff
@@ -146,7 +146,10 @@ SharedFrameMetricsHelper::UpdateFromCompositorFrameMetrics(
 {
   MOZ_ASSERT(aLayer);
 
-  CompositorChild* compositor = CompositorChild::Get();
+  CompositorChild* compositor = nullptr;
+  if (XRE_GetProcessType() != GeckoProcessType_Default) {
```

Comment:
For what it's worth I've been running mostly debug builds for the last week or so with most of those patches applied and I haven't seen any failing asserts or other problems. I don't know gecko well enough too be completely sure the patches don't break something, but so far I haven't see any major breakage.